plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -594,6 +614,23 @@ func (p *Plugin) admitServiceAccount(nodeName string, a admission.Attributes) er
 		return admission.NewForbidden(a, fmt.Errorf("node requested token bound to a pod scheduled on a different node"))
 	}
 
+	if p.serviceAccountNodeAudienceRestriction {
+		// ensure all items in tr.Spec.Audiences are present in a volume mount in the pod
+		requestedAudience := ""
+		switch len(tr.Spec.Audiences) {
+		case 0:
+			requestedAudience = ""
+		case 1:
+			requestedAudience = tr.Spec.Audiences[0]
+		default:
+			return admission.NewForbidden(a, fmt.Errorf("node may only request 0 or 1 audiences"))
```

Comment:
This works on pods because node projected volumes only can have one `aud`. 

TokenRequest for CSI drivers can specify multiple `tokenRequest` objects, each with one aud https://kubernetes-csi.github.io/docs/token-requests.html#example

I'm not familiar with how exactly CSI token request does that, it may be multiple tokens each with a single `aud`, which would work fine with this restriction.

Reply:
> I'm not familiar with how exactly CSI token request does that, it may be multiple tokens each with a single aud, which would work fine with this restriction.

that's right! kubelet requests a token for each audience configured in the `tokenRequests` and sends all the tokens to the CSI driver.

format of the tokens sent from kubelet->csi driver as part of the RPC call
```
	// Bound token is of the format:
	// "csi.storage.k8s.io/serviceAccount.tokens": {
	//  <audience>: {
	//    'token': <token>,
	//    'expirationTimestamp': <expiration timestamp in RFC3339 format>,
	//  },
	//  ...
	// }
```

Reply:
https://github.com/kubernetes/kubernetes/blob/632ed16e002d87fa7166a8f4ca1dc48d4f0a9725/pkg/volume/csi/csi_mounter.go#L372-L395